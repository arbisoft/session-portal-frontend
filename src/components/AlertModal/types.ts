export type AlertModalProps = {
  handleCloseAlertModal: () => void;
  errorMessage?: string;
  vertical?: "bottom" | "top";
  horizontal?: "center" | "left" | "right";
  severity?: "success" | "info" | "warning" | "error";
};
