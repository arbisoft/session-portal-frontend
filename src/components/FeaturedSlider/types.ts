import { ReactNode } from "react";

export type FeaturedSliderProps = {
  slides: ReactNode[];
};
