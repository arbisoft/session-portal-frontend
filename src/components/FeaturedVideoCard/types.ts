export type FeaturedVideoCardProps = {
  isVisible?: boolean;
  className?: string;
  date: string;
  imgUrl?: string;
  title: string;
  description: string;
  organizerName: string;
  width?: string;
};
