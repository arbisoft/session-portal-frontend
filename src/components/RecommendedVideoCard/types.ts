export interface RecommendedVideoCardProps {
  className?: string;
  date: string;
  duration: string;
  imgUrl: string;
  ratingValue: number;
  title: string;
  width?: string;
}
