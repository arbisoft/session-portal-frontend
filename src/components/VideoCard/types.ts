export type VideoCardProps = {
  className?: string;
  date: string;
  imgUrl?: string;
  title: string;
  organizerName: string;
  width?: string;
};
