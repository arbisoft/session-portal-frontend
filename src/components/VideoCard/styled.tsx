import Box from "@mui/material/Box";
import Card from "@mui/material/Card";
import { cardContentClasses } from "@mui/material/CardContent";
import { skeletonClasses } from "@mui/material/Skeleton";
import { styled, css, alpha } from "@mui/material/styles";
import { typographyClasses } from "@mui/material/Typography";

import { pxToRem, shouldForwardProp } from "@/utils/styleUtils";

export const VideoCardContainer = styled(Card, {
  name: "VideoCardContainer",
  shouldForwardProp,
})<{ $width: string; $height: string }>(({ theme, $width, $height }) => {
  return css`
    background: transparent;
    border-radius: unset;
    cursor: pointer;
    display: grid;
    overflow: unset;
    width: ${$width};
    height: ${$height};

    .${cardContentClasses.root} {
      align-items: flex-start;
      display: flex;
      flex-direction: column;
      gap: ${theme.spacing(10 / 8)};
      padding: 0;
      width: 100%;

      :last-child {
        padding-bottom: 0px;
      }

      img {
        border-radius: ${theme.shape.borderRadius + 8}px;
        height: auto;
        max-height: 192px;
        min-height: 192px;
        object-fit: cover;
        width: 100%;
      }

      .${skeletonClasses.root} {
        border-radius: ${theme.shape.borderRadius + 8}px;
        position: absolute;
        top: 0;
        z-index: -1;
        width: 100%;
        height: 100%;
      }

      .video-detail {
        display: flex;
        flex-direction: column;
        gap: 7px;
        width: 100%;

        .${typographyClasses.h1}, .${typographyClasses.h5}, .${typographyClasses.h3} {
          -webkit-box-orient: vertical;
          -webkit-line-clamp: 2;
          display: -webkit-box;
          overflow: hidden;
          text-overflow: ellipsis;
        }

        .date-time,
        .organizer-name {
          font-size: ${pxToRem(12)};
          font-weight: 500;
        }

        .video-description {
          -webkit-box-orient: vertical;
          -webkit-line-clamp: 4;
          display: -webkit-box;
          font-size: ${pxToRem(14)};
          font-weight: 500;
          overflow: hidden;

          ${theme.breakpoints.down("md")} {
            -webkit-line-clamp: 2;
          }
        }
      }
    }

    &.normal-card {
      .${cardContentClasses.root} {
        .video-detail {
          .${typographyClasses.h3} {
            font-size: ${pxToRem(16)};
            font-weight: 600;
          }
        }
      }
    }

    &.related-card {
      display: flex;
      margin-bottom: 20px;
      width: 100%;

      .${cardContentClasses.root} {
        flex-direction: row;
        gap: 10px;

        .image-wrapper {
          border-radius: ${theme.shape.borderRadius + 2}px;
          overflow: hidden;
          width: 113px;

          .video-player {
            border-radius: ${theme.shape.borderRadius + 2}px;
          }

          img {
            border-radius: ${theme.shape.borderRadius + 2}px;
            height: unset;
            max-height: unset;
            min-height: unset;
          }

          .${skeletonClasses.root} {
            border-radius: ${theme.shape.borderRadius + 2}px;
          }
        }

        .video-detail {
          gap: 4px;
          width: calc(100% - 113px);

          .${typographyClasses.h5} {
            font-size: ${pxToRem(12)};
            font-weight: 500;
          }

          .organizer-name,
          .date-time {
            font-size: ${pxToRem(12)};
          }
        }
      }
    }

    &.featured-card {
      background-color: ${alpha(theme.palette.common.white, theme.palette.mode === "dark" ? 0.06 : 0.7)};
      border-radius: 12px;
      display: flex;
      padding: 30px;

      .${cardContentClasses.root} {
        flex-direction: row;
        gap: ${theme.spacing(2.5)};

        ${theme.breakpoints.down("sm")} {
          flex-direction: column;
        }

        .image-wrapper {
          border-radius: ${theme.shape.borderRadius + 8}px;
          height: auto;
          object-fit: cover;
          width: 420px;

          ${theme.breakpoints.down("md")} {
            width: 300px;
          }

          ${theme.breakpoints.down("sm")} {
            width: 100%;
          }

          img {
            border-radius: ${theme.shape.borderRadius + 8}px;
            height: unset;
            max-height: unset;
            min-height: unset;
          }

          .${skeletonClasses.root} {
            border-radius: ${theme.shape.borderRadius + 8}px;
          }
        }

        .video-detail {
          width: calc(100% - 430px);
          gap: 5px;

          ${theme.breakpoints.down("md")} {
            width: calc(100% - 320px);
          }

          ${theme.breakpoints.down("sm")} {
            width: 100%;
          }

          .${typographyClasses.h1} {
            font-weight: 500;
          }

          .organizer-name,
          .date-time {
            font-size: ${pxToRem(14)};
          }
        }
      }
    }

    &.search-card {
      display: flex;

      .${cardContentClasses.root} {
        flex-direction: row;
        gap: ${theme.spacing(2.5)};

        ${theme.breakpoints.down("sm")} {
          flex-direction: column;
        }

        .image-wrapper {
          border-radius: ${theme.shape.borderRadius + 8}px;
          width: 400px;

          ${theme.breakpoints.down("md")} {
            width: 300px;
          }

          ${theme.breakpoints.down("sm")} {
            width: 100%;
          }

          img {
            border-radius: ${theme.shape.borderRadius + 8}px;
            height: unset;
            max-height: unset;
            min-height: unset;
          }

          .${skeletonClasses.root} {
            border-radius: ${theme.shape.borderRadius + 8}px;
          }
        }

        .video-detail {
          width: calc(100% - 430px);
          gap: 10px;

          ${theme.breakpoints.down("md")} {
            width: calc(100% - 320px);
          }

          ${theme.breakpoints.down("sm")} {
            width: 100%;
          }

          .${typographyClasses.h1} {
            font-weight: 500;
            font-size: ${pxToRem(20)};
          }

          .organizer-name,
          .date-time {
            font-size: ${pxToRem(14)};
          }
        }
      }
    }

    // for storybook
    &.custom-video-card {
      .${cardContentClasses.root} {
        .video-detail {
          * {
            color: ${theme.palette.secondary.main};
          }
        }
      }
    }
  `;
});

export const ImageWrapper = styled(Box, {
  name: "ImageWrapper",
})(({ theme }) => {
  return css`
    display: inline-flex;
    position: relative;
    width: 100%;

    .video-player {
      border-radius: ${theme.shape.borderRadius + 8}px;
      height: 100%;
      left: 0;
      object-fit: cover;
      position: absolute;
      top: 0;
      width: 100%;
      z-index: 2;
    }

    .video-duration {
      background-color: ${alpha(theme.palette.common.black, 0.7)};
      border-radius: 2px;
      color: ${theme.palette.common.white};
      font-size: ${pxToRem(12)};
      padding: 2px 4px;
      position: absolute;
      bottom: 5px;
      right: 5px;
      z-index: 1;
    }
  `;
});
