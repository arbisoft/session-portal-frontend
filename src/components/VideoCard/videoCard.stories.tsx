import type { Meta, StoryObj } from "@storybook/react";

import { VideoCardProps } from "./types";
import VideoCard from "./videoCard";

// Define the metadata for the component
const meta: Meta<VideoCardProps> = {
  title: "Components/RecommendedVideoCard", // Storybook title
  component: VideoCard, // The component itself
  tags: ["autodocs"], // Optional: Enable autodocs for better documentation
  argTypes: {
    // Define controls for the props
    title: {
      control: "text",
      description: "The title of the video",
    },
    organizerName: {
      control: "text",
      description: "The organizer/creator of the video",
    },
    date: {
      control: "text",
      description: "The upload date of the video",
    },
    imgUrl: {
      control: "text",
      description: "The URL of the video thumbnail image",
    },
    width: {
      control: "text",
      description: "The width of the card (e.g., '100%', '315px')",
    },
  },
};

export default meta;

// Define the template for the stories
type Story = StoryObj<VideoCardProps>;

// Default story
export const Default: Story = {
  args: {
    title: "Sample Video Title",
    organizerName: "Arbisoft",
    date: "2023-10-01",
    imgUrl: "/assets/images/temp-youtube-logo.webp",
  },
};

// Story with a long title
export const LongTitle: Story = {
  args: {
    ...Default.args,
    title: "This is a very long video title that should be truncated",
    width: "315px",
  },
};
