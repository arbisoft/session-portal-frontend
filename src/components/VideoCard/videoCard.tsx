"use client";

import React, { FC } from "react";

import Box from "@mui/material/Box";
import CardContent from "@mui/material/CardContent";
import Typography from "@mui/material/Typography";
import Image from "next/image";

import { VideoCardContainer } from "./styled";
import { VideoCardProps } from "./types";

const VideoCard: FC<VideoCardProps> = ({ className, date, imgUrl, organizerName, title, width = "315px" }) => {
  return (
    <VideoCardContainer $width={width} className={className} data-testid="video-card">
      <CardContent>
        <Image alt={title} height={192} width={315} src={imgUrl ?? "/assets/images/temp-youtube-logo.webp"} />
        <Box className="video-detail">
          <Typography variant="h3" component="div" title={title}>
            {title}
          </Typography>
          <Typography variant="body2" className="organizer-name" data-testid="video-card-organizer">
            {organizerName}
          </Typography>
          <Typography variant="body2" className="date-time" data-testid="video-card-date-time">
            {date}
          </Typography>
        </Box>
      </CardContent>
    </VideoCardContainer>
  );
};

export default VideoCard;
