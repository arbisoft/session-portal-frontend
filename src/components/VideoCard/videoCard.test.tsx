import { render, screen } from "@testing-library/react";

import { VideoCardProps } from "./types";
import VideoCard from "./videoCard";

const mockProps: VideoCardProps = {
  className: "custom-class",
  date: "Jan 01, 2024",
  imgUrl: "https://example.com/sample.jpg",
  title: "Sample Video Title",
  organizerName: "Sample Video Organizer",
  width: "300px",
};

describe("VideoCard", () => {
  test("should renders the component with provided props", () => {
    render(<VideoCard {...mockProps} />);

    expect(screen.getByText(mockProps.title)).toBeInTheDocument();
    expect(screen.getByTestId("video-card-date-time")).toBeInTheDocument();
    const imgUrl = screen.getByRole("img", { name: mockProps.title }).getAttribute("src") ?? "";
    expect(screen.getByRole("img", { name: mockProps.title })).toHaveAttribute("src", imgUrl);
  });

  test("should displays default image when imgUrl is not provided", () => {
    render(<VideoCard {...mockProps} imgUrl={undefined} />);

    const imgUrl = screen.getByRole("img", { name: mockProps.title }).getAttribute("src") ?? "";

    expect(decodeURIComponent(imgUrl)).toContain("/assets/images/temp-youtube-logo.webp");
  });

  test("should renders the organizer name", () => {
    render(<VideoCard {...mockProps} />);
    expect(screen.getByText(mockProps.title)).toBeInTheDocument();
    expect(screen.getByTestId("video-card-organizer")).toBeInTheDocument();
  });

  test("should matches snapshot", () => {
    const { asFragment } = render(<VideoCard {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
