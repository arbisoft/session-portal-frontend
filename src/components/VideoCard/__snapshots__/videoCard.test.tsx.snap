// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoCard should matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root custom-class normal-card css-1a4wa29-MuiPaper-root-MuiCard-root-VideoCardContainer-root"
    data-testid="video-card"
    style="--Paper-shadow: var(--mui-shadows-0); --Paper-overlay: var(--mui-overlays-0);"
  >
    <div
      class="MuiCardContent-root css-1lt5qva-MuiCardContent-root"
    >
      <div
        class="image-wrapper MuiBox-root css-1k95zkw-ImageWrapper-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-wave css-116i1gn-MuiSkeleton-root"
          style="width: 100%; height: 100%;"
        />
        <img
          alt="Refresher & AMA Session on Competency Framework Changes - January 2025"
          data-testid="video-card-image"
          height="196"
          src="http://localhost:1234/media/thumbnails/Screenshot_2025-02-12_at_1.13.39PM.png"
          width="315"
        />
        <div
          class="MuiTypography-root MuiTypography-bodyMedium video-duration css-1xa7gk3-MuiTypography-root"
          data-testid="video-card-duration"
        >
          30:30
        </div>
      </div>
      <div
        class="video-detail MuiBox-root css-0"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 css-1tq2yg0-MuiTypography-root"
          data-testid="video-card-title"
          title="Refresher & AMA Session on Competency Framework Changes - January 2025"
        >
          Refresher & AMA Session on Competency Framework Changes - January 2025
        </h3>
        <div
          class="MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-bodyMedium organizer-name css-l4hfz4-MuiTypography-root"
            data-testid="video-card-organizer"
          >
            John Doe
          </p>
          <p
            class="MuiTypography-root MuiTypography-bodyMedium date-time css-l4hfz4-MuiTypography-root"
            data-testid="video-card-date-time"
          >
            Jan 09, 2025
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
